model/openapi.yml

```diff
@@ -139,6 +139,13 @@ components:
         - type
         - metadata
       properties:
+        id:
+          type: string
+          description: |
+            The ID of the agent.
+            This is the ID that will be used to reference this agent, if no ID is passed one will be generated.
+            If another agent is enrolled with the same ID the other agent will no longer be able to communicate,
+            this new agent is considered a replacement of the other agent.
```

Comment:
This is a pretty big change from our discussion.
Please also add a sentence saying the (replaced) agent will still be able to send data into ES

Reply:
It is a slight change, because it is not possible to get an API key token again after the initial create. That made me have no choice to change the behavior.

Reply:
I did add more as requested to the description of this field to inform that it will still allow data to flow.